src/overlays/actors/ovl_Bg_Gnd_Firemeiro/z_bg_gnd_firemeiro.c

```diff
@@ -35,17 +31,122 @@ const ActorInit Bg_Gnd_Firemeiro_InitVars = {
     (ActorFunc)BgGndFiremeiro_Update,
     (ActorFunc)BgGndFiremeiro_Draw,
 };
-*/
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Bg_Gnd_Firemeiro/BgGndFiremeiro_Init.s")
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Bg_Gnd_Firemeiro/BgGndFiremeiro_Destroy.s")
+void BgGndFiremeiro_Init(Actor* thisx, GlobalContext* globalCtx) {
+    s32 pad;
+    BgGndFiremeiro* this = THIS;
+    CollisionHeader* colHeader = NULL;
+
+    ActorShape_Init(&this->dyna.actor.shape, 0.0f, NULL, 0.0f);
+    Actor_SetScale(&this->dyna.actor, 0.1f);
+    this->initPos = this->dyna.actor.world.pos;
+
+    if (this->dyna.actor.params == 0) {
+        DynaPolyActor_Init(&this->dyna, DPM_UNK);
+        CollisionHeader_GetVirtual(&gFireTrialPlatformCol, &colHeader);
+        this->dyna.bgId = DynaPoly_SetBgActor(globalCtx, &globalCtx->colCtx.dyna, &this->dyna.actor, colHeader);
+        this->actionFunc = BgGndFiremeiro_Rise;
+    }
+}
+
+void BgGndFiremeiro_Destroy(Actor* thisx, GlobalContext* globalCtx2) {
+    GlobalContext* globalCtx = globalCtx2;
+    BgGndFiremeiro* this = THIS;
+
+    if (this->dyna.actor.params == 0) {
+        if (1) {}
+        DynaPoly_DeleteBgActor(globalCtx, &globalCtx->colCtx.dyna, this->dyna.bgId);
+    }
+}
+
+void BgGndFiremeiro_Sink(BgGndFiremeiro* this, GlobalContext* globalCtx) {
+    f32 sunkHeight = this->initPos.y - 150.0f;
+
+    if (func_8004356C(&this->dyna)) {
+        this->timer = 10;
+    }
+
+    if (sunkHeight < this->dyna.actor.world.pos.y) {
+        this->dyna.actor.world.pos.y -= 0.5f;
+
+        if (this->dyna.actor.world.pos.y < sunkHeight) {
+            this->dyna.actor.world.pos.y = sunkHeight;
+        }
+
+        func_8002F948(&this->dyna.actor, NA_SE_EV_ROLL_STAND_2 - SFX_FLAG);
+    }
+
+    if (this->timer > 0) {
+        this->timer--;
+    } else {
+        this->actionFunc = BgGndFiremeiro_Rise;
+    }
+}
+
+void BgGndFiremeiro_Shake(BgGndFiremeiro* this, GlobalContext* globalCtx) {
+    s32 pad;
+    f32 randSign;
+
+    if (func_8004356C(&this->dyna)) { // Player standing on it
+        if (this->timer > 0) {
+            this->timer--;
+
+            randSign = ((this->timer & 1) ? 2.0f : -2.0f);
```

Comment:
```suggestion
            randSign = (this->timer & 1) ? 2.0f : -2.0f;
```

Reply:
I like my ternaries in brackets for clarity's sake. Can change it if it's contrary to our style.

Reply:
I dont think we have a set style but IMO it looks nicer without them